app/src/main/java/com/kevinlu/watstats/MainActivity.java

```diff
@@ -1,13 +1,164 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.francochen.watcard.WatCardAPI;
+import com.francochen.watcard.WatCardService;
+import com.francochen.watcard.model.RequestVerificationToken;
+import com.francochen.watcard.model.balance.BalanceType;
+import com.francochen.watcard.model.balance.Balances;
+import com.kevinlu.watstats.models.AccountBalance;
+import com.kevinlu.watstats.models.MonthlySpending;
+
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class MainActivity extends AppCompatActivity {
 
+    private TextView totalBalance;
+    private ViewPager accountBalances;
+    private ViewPager monthlySpendings;
+    private AccountBalanceAdapter accountBalanceAdapter;
+    private MonthlySpendAdapter monthlySpendAdapter;
+    private List<AccountBalance> accountBalanceList;
+    private List<MonthlySpending> monthlySpendingList;
+    private WatCardAPI api = new WatCardAPI();
+    private WatCardService service = api.createService();
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        totalBalance = findViewById(R.id.total_balance);
+
+        accountBalanceList = new ArrayList<>();
+
+        accountBalanceAdapter = new AccountBalanceAdapter(accountBalanceList, this);
+
+        accountBalances = findViewById(R.id.account_balances);
+        accountBalances.setAdapter(accountBalanceAdapter);
+        accountBalances.setPadding(dpToPx(36), dpToPx(36), dpToPx(205), 0);
+
+        accountBalances.addOnPageChangeListener(new ViewPager.SimpleOnPageChangeListener() {
+            @Override
+            public void onPageSelected(int position) {
+                if (position < accountBalanceList.size() - 1) {
+                    return;
+                }
+                accountBalances.setCurrentItem(position - 1, true);
+            }
+        });
+
+        monthlySpendingList = new ArrayList<>();
+        monthlySpendingList.add(new MonthlySpending("December", "$ 254.98"));
+        monthlySpendingList.add(new MonthlySpending("November", "$ 392.11"));
+        monthlySpendingList.add(new MonthlySpending("October", "$ 153.50"));
+
+        monthlySpendAdapter = new MonthlySpendAdapter(monthlySpendingList, this);
+
+        monthlySpendings = findViewById(R.id.monthly_spend_pager);
+        monthlySpendings.setAdapter(monthlySpendAdapter);
+        monthlySpendings.setPadding(dpToPx(18), 0, dpToPx(226), 0);
+
+        monthlySpendings.addOnPageChangeListener(new ViewPager.SimpleOnPageChangeListener() {
+            @Override
+            public void onPageSelected(int position) {
+                if (position < monthlySpendingList.size() - 1) {
+                    return;
+                }
+                monthlySpendings.setCurrentItem(position - 1, true);
+            }
+        });
+
+        login("user", "pass");
```

Comment:
If you are not ready for the login feature, maybe just remove it.

Reply:
The login is working, I just don't have a LoginActivity for it yet.